### PR TITLE
Jetpack Search: update upsell links to go directly to checkout.

### PR DESCRIPTION
**Pull request by gibrown**
Going direct seems to be a better UX.

#### Testing instructions

There are two places these links show up for sites that don't have a JPS subscription:
* The Jetpack>Search page
* The Settings>Performance page

There is a bug in the nav unification that makes testing the Jetpack>Search page tough: https://github.com/Automattic/wp-calypso/issues/49345

### CODE REVIEW

**Comment by bluefuton** [comment]
> The Settings>Performance page

Works well here 👍 

I wasn't able to test the Jetpack>Search page because my purchases aren't being loaded. I'll look into that.

**Review by jsnmoon — APPROVED** [review]
Feels much better 👍. 

As @bluefuton mentioned, I was also unable to test `Jetpack>Search`. However, I'm pretty certain that this change will work as expected. 

After #49380 ships, I recommend rebasing this PR, smoke testing the change on `Jetpack>Search`, and then shipping without waiting on another review.

**Pull request merged**
